src/components/BookList/BookList.js

```diff
@@ -0,0 +1,202 @@
+import React, { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { APIS } from '../../config';
+import * as s from '../../pages/ProductDetail/ProductDetailStyled';
+
+const BookList = ({ title, bookList }) => {
+  const [checkedList, setCheckedList] = useState([]);
+  const [showAllBooks, setShowAllBooks] = useState(false);
+
+  const visibleBooks = showAllBooks ? bookList : bookList.slice(0, 7);
+
+  const handleShowMore = () => {
+    setShowAllBooks(prev => !prev);
+  };
+
+  const navigate = useNavigate();
+
+  const handleSingleCheck = (checked, id) => {
+    if (checked) {
+      setCheckedList(prev => [...prev, id]);
+    } else {
+      setCheckedList(checkedList.filter(el => el !== id));
+    }
+  };
+
+  const handleAllCheck = checked => {
+    if (checked) {
+      const idArr = [];
+      bookList.forEach(el => {
+        idArr.push(el.id);
+      });
+      setCheckedList(idArr);
+    } else {
+      setCheckedList([]);
+    }
+  };
+
+  const clickToOrder = e => {
+    fetch(APIS.owner, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json;charset=utf-8',
+        Authorization: localStorage.getItem('TOKEN'),
+      },
+      body: JSON.stringify({
+        type: 'owner_price',
+        bookId: checkedList,
+        // checkedList의 각 권의 id(bookId)
+      }),
+    })
+      .then(response => response.json())
+      .then(data => {
+        if (data.message === 'NOT_EXIST_TOKEN') {
+          alert('로그인이 필요합니다');
+        } else if (data.message === '선택된 도서가 없습니다.') {
+          alert('선택된 도서가 없습니다.');
+        } else {
+          alert('결제가 완료되었습니다.');
+          navigate('/bookcase');
+        }
```

Comment:
```suggestion
        if (data.message === 'success') {
          alert('결제가 완료되었습니다.');
          navigate('/bookcase');
        } else {
          alert(errorMsg[data.message]);
        }
```
이런식으로 분기처리를 하면 여러 에러메세지에 대해서도 깔끔하게 대응할 수 있겠네요!
만약 통신 성공했을 때 메세지가 success가 아닌 다른 값일 수도 있으니 확인하고
해당 코드 변경해주세요!